templates/data-provider/index.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useReducer } from 'react'
+import { TextField, Button } from '@oacore/design'
+import { useRouter } from 'next/router'
+
+import styles from './styles.module.css'
+import debounce from '../../utils/debounce'
+
+const initialState = {
+  helper:
+    'It can be any URL: homepage address, any data resource address, OAI-PMH endpoint or RIOXX endpoint',
+  form: {
+    dataProviderUrl: '',
+  },
+}
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'CLAIM_DATA_PROVIDER':
+      return state
+    case 'NEW_DATA_PROVIDER':
+      return state
+    case 'DATA_PROVIDER_NOT_FOUND':
+      return state
+    case 'CHANGE_FORM_DATA':
+      return {
+        form: {
+          ...state.form,
+          ...action.value,
+        },
+      }
+    default:
+      throw new Error()
+  }
+}
+
+const DataProviderPageTemplate = () => {
+  const [
+    {
+      helper,
+      form: { dataProviderUrl },
+    },
+    dispatch,
+  ] = useReducer(reducer, initialState)
+  const router = useRouter()
+  const handleChange = debounce((event) => {
+    dispatch({
+      type: 'CHANGE_FORM_DATA',
+      value: { [event.target.name]: event.target.value },
+    })
+  })
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    // API call here probably
+    // Redirect user to the next step
+    router.push('/data-provider-details')
+  }
+
+  useEffect(() => {
+    // check if data provider exist
+    // dispatch({
+    //   type: 'DATA_PROVIDER_NOT_FOUND',
+    //   value: {},
+    // })
+    // dispatch({
+    //   type: 'NEW_DATA_PROVIDER',
+    //   value: {},
+    // })
+    // dispatch({
+    //   type: 'CLAIM_DATA_PROVIDER',
+    //   value: {},
+    // })
+  }, [dataProviderUrl])
+
+  return (
+    <form onChange={handleChange} onSubmit={handleSubmit}>
```

Comment:
This render method should stay pure. The state management should be placed in another component. I create a task for it.

For this moment I propose to remove state management at all and add `defaultValue` prop instead.